File: LeetCode/0006.Zigzag-Conversion/Zigzag-Conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if(s == null || numRows <= 1) return s;
        StringBuilder [] sb = new StringBuilder[numRows];
        for(int i = 0; i < numRows; i++){
            sb[i] = new StringBuilder();
        }
        for(int i = 0; i < s.length(); i++){
            int index = i % (2 * numRows - 2);
            int row = (index < numRows)? index: 2 * numRows - 2 - index;
            sb[row].append(s.charAt(i));
        }
        for(int i = 1; i < numRows; i++){
            sb[0].append(sb[i]);
        }
        return sb[0].toString();
    }
}

File: LeetCode/0006.Zigzag-Conversion/Zigzag-Conversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1: return s
        res = ""
        n = len(s)
        for i in range(numRows):
            j, k = 0, numRows + numRows - 2
            while j * k + i < n:
                res += s[j * k + i]
                if i != 0 and i != numRows - 1 and j * k + k - i < n:
                    res += s[j * k + k - i]
                j += 1
        return res
                
        
